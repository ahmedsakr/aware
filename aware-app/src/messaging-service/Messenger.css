.Messenger {
  text-align: center;
  display: block;
}

.aware-container {
  height: 100vh;
  max-height: 100vh;
}

.aware-column {
  height: 100%;
}
