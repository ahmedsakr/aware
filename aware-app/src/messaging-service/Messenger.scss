/*
 * Messenger styling
 *
 * Defines layout fundamentals for the Messenger.
 */

.aware-container {
    height: 100vh;
    max-height: 100vh;
}

.aware-column {
    height: 100%;
}

#messenger-root {
    flex-direction: column;
    height: 100%;
}

#navigation-header {
    flex: 0 1 auto;
}

#messenger-body {
    overflow: auto;
    font-family: 'Roboto';
}

#messenger {
    display: flex;
    flex-flow: column;
    justify-items: stretch;
}