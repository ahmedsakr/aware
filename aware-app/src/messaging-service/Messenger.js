import React, { Component } from 'react';
import './Messenger.scss';
import '../../node_modules/bootstrap/dist/css/bootstrap.css'
import '../../node_modules/font-awesome/css/font-awesome.min.css'

import NavBar from './NavigationBar/NavBar/NavBar'
import ChatSelector from './ChatSelector/ChatSelector'
import ChatWindow from './ChatFeature/ChatWindow/ChatWindow'
import ActivityPanel from './ChatFeature/ActivityPanel/ActivityPanel'
import MessageInput from './ChatFeature/MessageInput/MessageInput'
import NewsletterOverlay from '../shared/overlay/test/NewsletterOverlay'

class Messenger extends Component {
  constructor() {
    super();

    this.state = {
      messages: [],
      chatTitle: ""
    }
  }

  componentDidMount() {
    if (this.props.socket) {
      this.props.socket.on('chat message', message => {
        this.setState({
          messages: this.state.messages.concat([message])
        })
      })

      this.props.socket.on('chat history', messages => {
        this.setState({
          messages: messages
        })
      })
    }
  }

  render() {
    const {selectRoom, sendMessage } = this;

    return (
      <div class="aware-container" className="App">
        <div className="container-fluid aware-container">
          <NewsletterOverlay />

          <div id="messenger-root" class="row">
            <div class="col-12 p-0" id="navigation-header">
              <NavBar activeRoom={this.state.chatTitle} />
            </div>

            <div class="col-12 p-0" id="messenger-body">
              <div class="col-2 p-0">
                <ChatSelector
                  socket={this.props.socket}
                  username={this.props.username}
                  selectRoom={selectRoom} />
              </div>

              <div id="messenger" class="col-10 p-0">
                <ActivityPanel />

                <ChatWindow
                  messages={this.state.messages}
                  name={this.props.username} />

                <MessageInput
                  sendMessage={sendMessage}
                  name={this.props.username} />
              </div>
            </div>
          </div>
        </div>
        
      </div>
    );
  }

  selectRoom = (groupId, title) => {
    this.props.socket.emit('room', groupId)
    this.setState({ 
      messages: [],
      chatTitle: title,
      groupId: groupId
     });
  }

  sendMessage = (message) => {
    this.props.socket.emit('chat message', message, this.state.groupId, this.props.username);
  }
}

export default Messenger;
