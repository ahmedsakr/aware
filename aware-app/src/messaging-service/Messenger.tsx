import React from 'react';
import './Messenger.scss';
import '../../node_modules/bootstrap/dist/css/bootstrap.css'
import '../../node_modules/font-awesome/css/font-awesome.min.css'

import NavBar from './NavigationBar/NavBar/NavBar'
import ChatSelector from './ChatSelector/ChatSelector'
import ChatWindow from './ChatFeature/ChatWindow/ChatWindow'
import { UserMessageContents } from './ChatFeature/ChatWindow/Message/Message'
import ActivityPanel from './ChatFeature/ActivityPanel/ActivityPanel'
import MessageInput from './ChatFeature/MessageInput/MessageInput'
import NewsletterOverlay from '../shared/overlay/test/NewsletterOverlay'

type MessengerProps = {
    socket: SocketIOClient.Socket,
    username: string
};

type MessengerState = {
    messages: UserMessageContents[],
    chatTitle: string,
    groupId: string
};

export default class Messenger extends React.Component<MessengerProps, MessengerState> {
    constructor(props: MessengerProps) {
        super(props);

        this.state = {
            messages: [],
            chatTitle: "",
            groupId: ""
        }
    }

    componentDidMount(): void {
        if (this.props.socket) {
            this.props.socket.on('chat message', (message: UserMessageContents) => {
                this.setState({
                    messages: this.state.messages.concat([message])
                })
            })

            this.props.socket.on('chat history', (messages: UserMessageContents[]) => {
                this.setState({
                    messages: messages
                })
            })
        }
    }

    render(): JSX.Element {
        const { selectRoom, sendMessage } = this;

        return (
            <div className="aware-container App">
                <div className="container-fluid aware-container">
                    <NewsletterOverlay />

                    <div id="messenger-root" className="row">
                        <div className="col-12 p-0" id="navigation-header">
                            <NavBar activeRoom={this.state.chatTitle} />
                        </div>

                        <div className="col-12 p-0" id="messenger-body">
                            <div className="col-2 p-0">
                                <ChatSelector
                                    socket={this.props.socket}
                                    username={this.props.username}
                                    selectRoom={selectRoom} />
                            </div>

                            <div id="messenger" className="col-10 p-0">
                                <ActivityPanel 
                                    socket={this.props.socket}
                                    activeRoom={'sysc2100'}/>

                                <ChatWindow
                                    messages={this.state.messages}
                                    name={this.props.username} />

                                <MessageInput
                                    sendMessage={sendMessage}
                                    name={this.props.username} />
                            </div>
                        </div>
                    </div>
                </div>

            </div>
        );
    }

    selectRoom = (groupId: string, title: string) => {
        this.props.socket.emit('room', groupId)
        this.setState({
            messages: [],
            chatTitle: title,
            groupId: groupId
        });
    }

    sendMessage = (message: UserMessageContents) => {
        this.props.socket.emit('chat message', message, this.state.groupId, this.props.username);
    }
}
