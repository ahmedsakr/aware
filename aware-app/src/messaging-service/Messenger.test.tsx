import React from 'react';
import { shallow } from 'enzyme';

import Messenger from './Messenger';
import io from 'socket.io-client'
jest.mock('socket.io-client');

describe('Messenger', () => {
    it('should render correctly in "debug" mode', () => {
        
        const channel = io();

        const component = shallow(
            <Messenger
                socket={channel}
                username="myusername" />
        );

        expect(component).toMatchSnapshot();
    });
});