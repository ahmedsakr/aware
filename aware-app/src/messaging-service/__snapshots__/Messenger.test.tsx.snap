version https://git-lfs.github.com/spec/v1
oid sha256:3127330939cc2d0747ad93c78b44d861a306a9b7fc2499b7497d9146f1787407
size 1068
